internal/chain/engine.go

```diff
@@ -190,44 +206,30 @@ func (e *engineImpl) VerifyVRF(
 
 // GetLeaderPubKeyFromCoinbase retrieve corresponding blsPublicKey from Coinbase Address
 func GetLeaderPubKeyFromCoinbase(
-	blockchain engine.ChainReader, h *block.Header,
+	slots shard.SlotList, coinbase common.Address, isStaking bool,
 ) (*bls.PublicKeyWrapper, error) {
-	shardState, err := blockchain.ReadShardState(h.Epoch())
-	if err != nil {
-		return nil, errors.Wrapf(err, "cannot read shard state %v %s",
-			h.Epoch(),
-			h.Coinbase().Hash().Hex(),
-		)
-	}
-
-	committee, err := shardState.FindCommitteeByID(h.ShardID())
-	if err != nil {
-		return nil, err
-	}
```

Comment:
Can you explain why removing that code helps with anything ?

Reply:
moved to another pr, those changes does not affect view change directly